src/utils/davix_logger.cpp

```diff
@@ -26,13 +26,34 @@
 #include <utils/davix_logger.hpp>
 #include <utils/stringutils.hpp>
 
+// This bit of code allows one to set the debug level via an envar when
+// the Davix library is being used via a plugin.
+//
+namespace
+{
+int getTraceValue()
+{
+   const char *dbg = getenv("DAVIX_DEBUG");
+   if (dbg)
+      {char dbgval = tolower(*dbg);
+       if (dbgval == 'c') return DAVIX_LOG_CRITICAL;
+       if (dbgval == 'w') return DAVIX_LOG_WARNING;
+       if (dbgval == 'v') return DAVIX_LOG_VERBOSE;
+       if (dbgval == 'd') return DAVIX_LOG_DEBUG;
+       if (dbgval == 't') return DAVIX_LOG_TRACE;
+       if (dbgval == 'a') return DAVIX_LOG_ALL;
+      }
+   return 0;
+}
+}
+
 #ifdef HAVE_ATOMIC
 #include <atomic>
-static std::atomic<int> internal_log_level(0);
+static std::atomic<int> internal_log_level(getTraceValue());
```

Comment:
Aha, you've promised a change that will only be triggered for the `DAVPOSIX_MPUPLOAD` scenario, but this here will always be executed and can be quite dangerous!

Reply:
This is the only case where I broke my promise because I was fxing a Davix deficieny. That is it is unable to turn on tracing unless you odify the program being traced or go through a contortion of gdb breakpoints and assigns. This code is no more dangerous thhan the code it replaces. By default it is equivalent to the code it replaces. If you set a debugging envar then you can select the debugging level before the priograms starts. A very useful addition. If you think this is not generally useful, I can move the set to the code controlled by the DAVPOSIX_MPUPLOAD  envar though that would pretty much make it only useful for me and you would have to beat your head against the wall in other scenarios.

Reply:
Just making a bit of drama. The function looks safe and default behavior would not change. The modification can stay.